src/Main.hs

```diff
@@ -206,8 +214,8 @@ versionOption = infoOption
 
 overrideConfig :: EConfig -> Options -> IO EConfig
 overrideConfig config Options{..} = do
-  rpcUrl <- Onchain.rpcUrlEnv
-  rpcBlock <- Onchain.rpcBlockEnv
+  rpcUrl <- (pure cliRpcUrl) <|> Onchain.rpcUrlEnv
```

Comment:
This unfortunately won't work since <|> here operates on IO, not on Maybe. Can you rename `rpcUrl` to `envRpcUrl` (same for block) and change this instead: `rpcUrl = cliRpcUrl <|> envRpcUrl <|> config.rpcUrl`

Reply:
Updated. Thanks for the quick review!